### PR TITLE
Fix defining multiple methods with the same block 

### LINKED ISSUE #497: define_method $block using all the same super
```rb
$block = proc {
  super()
}
class Base
   def foo
     puts "1"
   end
   def bar
    puts "2"
   end
end

class Derive < Base
  define_method :foo, $block
  define_method :bar, $block

end

d = Derive.new
d.foo #1
d.bar #2
# but opal inputs 2 in all two cases.
```

### PR DESCRIPTION
**Pull request by elia**
Fixes #497

### CODE REVIEW

File: opal/corelib/module.rb [hunk 1]
```diff
@@ -395,14 +395,27 @@ def define_method(name, method = undefined, &block)
               end
 
     %x{
-      var id = '$' + name;
+      // Wrapping and forwarding is required to make super work, as the
+      // same block can be used to define multiple methods.
+      const wrapper = function() {
```
Comment (hmdne) [inline opal/corelib/module.rb:400]: Can we not break the compatibility yet and do `var`, not `const`?

File: opal/corelib/module.rb [hunk 2]
```diff
@@ -395,14 +396,27 @@ def define_method(name, method = undefined, &block)
...
+        block.$$jsid        = name
+        block.$$p           = wrapper.$$p
+        block.$$s           = null;
+        block.$$def         = wrapper;
+        block.$$define_meth = true;
+
+        return block.apply(this, arguments)
+      }
+
+      var id = `$${name}`
```
Comment (hmdne) [inline opal/corelib/module.rb:411]: This is another thing I haven't noticed before. I would rather if it was '$'+name...

Comment (hmdne) [reply opal/corelib/module.rb:411]: Also, the linter has noticed it, but didn't fail the job:

```
* Parsing error: Unexpected character '`'
  - Compiled: tmp/lint/opal.js:3511:16
  - Original: corelib/module.rb:411:0
```

Comment (elia) [reply opal/corelib/module.rb:411]: sorry did that out of habit, will fix it along with the un-failed job 👍

**Review by hmdne — APPROVED** [review]

**Comment by hmdne** [comment]
After a deeper look, I don't think it will be wrong to backport it to stable :D

**Pull request merged**